database/sqlite3/sqlite3_test.go

```diff
@@ -181,3 +182,45 @@ func TestMigrateWithDirectoryNameContainsWhitespaces(t *testing.T) {
 	}
 	dt.Test(t, d, []byte("CREATE TABLE t (Qty int, Name string);"))
 }
+
+func TestDbPathFromURLWithSimpleURL(t *testing.T) {
```

Comment:
Combine theses 3 tests into a single [table driven test](https://github.com/golang/go/wiki/TableDrivenTests)

Also add these 2 test cases:
* % encoded character in filename
* % encoded character in query param

Reply:
Changed over the testing to follow the table driven test style, and added in the additional test cases requested.